packages/core/helper-plugin/lib/src/hooks/useCustomFields/useCustomFields.stories.mdx

```diff
@@ -0,0 +1,83 @@
+<!--- useCustomFields.stories.mdx --->
+
+import { Meta } from '@storybook/addon-docs';
+
+<Meta title="hooks/useCustomFields" />
+
+# useCustomFields
+
+This hook provides you with a `get` function to retrieve information about a specific custom field registered in the app, as well as an `getAll` function to obtain information about all custom fields registered in the app.
+
+## Usage
+
+### Get information about a specific plugin if I know its uid
+
+```js
+import { useCustomFields } from '@strapi/helper-plugin';
+
+const CustomIcon = ({ customFieldUuid }) => {
+  const customFieldsRegistry = useCustomFields();
+  const customField = customFieldsRegistry.get(customFieldUuid);
+
+  if (customField?.icon) {
+    return customFieldIcon;
+  }
+
+  return <Placeholder />;
+};
+```
+
+### Get all the custom fields
+
+```js
+import { useCustomFields } from '@strapi/helper-plugin';
+import { Typography } from '@strapi/design-system';
+
+const CustomFieldsList = () => {
+  const customFieldsRegistry = useCustomFields();
+  const customFields = customFieldsRegistry.getAll();
+  const registeredCustomFields = Object.entries(customFields.getAll());
+
+  return (
+    <>
+      {registeredCustomFields.map(([uid, customField]) => (
+        <Typography>{`${customField.name} uid: ${uid}`}</Typography>
+      ))}
+    </>
+  );
+};
+```
+
+## Methods
+
+### `get(customFieldUid: Uid): Object`
+
+With the method from `useCustomFields` hook you can have all the information of a custom field.
+
+### `getAll(): Object`
+
+With this method from `useCustomFields` hook, you will receive a dictionary containing all the custom fields with their respective information.
+
+## Typescript
+
+```ts
+interface CustomField {
+  components: Object;
+  icon: React.ComponentType;
```

Comment:
```suggestion
  icon: React.ComponentType;
```

Do you know how we render `icon`? Do we have to do `<Icon />` or can someone just do `<>{icon}</>`, if it's the latter we should do `React.ReactNode` 👍🏼

Reply:
Great, I think you're correct with your type :)